test/e2e/lua/dynamic_configuration.go

```diff
@@ -104,6 +112,30 @@ var _ = framework.IngressNginxDescribe("Dynamic Configuration", func() {
 			Expect(restOfLogs).ToNot(ContainSubstring("first sync of Nginx configuration"))
 		})
 
+		It("configuration module should read from temp file when request body > client_body_buffer_size", func() {
+			// Update client-body-buffer-size to 1 byte
+			err := f.UpdateNginxConfigMapData("client-body-buffer-size", "8")
```

Comment:
lower than 8 is not possible?

Reply:
Didn't really make a difference, I could lower it though. 

My hypothesis for why it doesn't matter comes from reading about [how nginx applies the client body buffer size limit](https://github.com/nginx/nginx/blob/55b37eff8f700ea1dd279368d5a4a7b00f3c1344/src/http/ngx_http_request_body.c#L88). Seems like there is sometimes Nginx pre-reads responses, and if Nginx is able to pre-read the entire response in the pre-read buffer - the client body buffer size is not looked at at all.

It also seems to compare only what is left of the response after pre-reading against the client body buffer size instead of the size of the entire response: https://github.com/nginx/nginx/blob/55b37eff8f700ea1dd279368d5a4a7b00f3c1344/src/http/ngx_http_request_body.c#L165

But yes, can lower it to 1/2 for clarity. Seems like we would still need the same amount of replicas though to reliably reproduce temp file buffering.